controllers/barbicanapi_controller.go

```diff
@@ -936,7 +867,41 @@ func (r *BarbicanAPIReconciler) reconcileNormal(ctx context.Context, instance *b
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *BarbicanAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *BarbicanAPIReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
+	Log := r.GetLogger(ctx)
+	secretFn := func(_ context.Context, o client.Object) []reconcile.Request {
```

Comment:
not tested, but I think we could just index it like we do with the passwordSecret and such. 

```
	// index customServiceConfigSecrets
	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &barbicanv1beta1.BarbicanAPI{}, customServiceConfigSecretsField, func(rawObj client.Object) []string {
		// Extract the customServiceConfigSecrets names from the spec, if one is provided
		cr := rawObj.(*barbicanv1beta1.BarbicanAPI)
		
		return cr.Spec.CustomServiceConfigSecrets
	}); err != nil {
		return err
	}
```